tools/memap.py

```diff
@@ -205,18 +253,41 @@ def parse_map_file_gcc(self, file_desc):
                 elif change_section != False:
                     current_section = change_section
 
-                [module_name, module_size] = self.parse_section_gcc(line)
+                [object_name, object_size] = self.parse_section_gcc(line)
 
-                if module_size == 0 or module_name == "":
+                if object_size == 0 or object_name == "":
                     pass
                 else:
-                    self.module_add(module_name, module_size, current_section)
+                    self.module_add(object_name, object_size,\
+                                        current_section)
+
+    def parse_object_name_armcc(self, line):
+        """ Parse object file
+
+        Positional arguments:
+        line - the line containing the object or library
+        """
+
+        # simple object (not library)
+        if line[-3] == '.' and line[-2] == 'o':
+            object_name = line[:-1]
+            return object_name
+
+        else:
+
+            rex_obj_name = r'(.+\.l)\((.+\.o)\)'
+            test_rex_obj_name = re.match(rex_obj_name, line)
+
+            if test_rex_obj_name:
+                object_name = test_rex_obj_name.group(1) + '/' + \
+                              test_rex_obj_name.group(2)
+
+                return '[lib]/' + object_name
+            else:
+                print "BUG armcc map parser (1): " + line
```

Comment:
I don't think this is a bug. Malformed input should be thought of as a bug outside this program, not in it. Could you change this text to reflect that? Something like "malformed input found when parsing armcc map: %s" % line would be fine.